src/app/components/rateProduct.js

```diff
@@ -3,32 +3,37 @@ import notify from './../base/notify';
 import { notes } from './../constants/notes';
 import { messages } from './../constants/messages';
 
-export default () => {
-    const APIURl = 'http://localhost:5003/rating';
-    const rateBtn = document.querySelector('.rate-product__btn');
-    const reviewText = document.querySelector('.rate-product__input');
-    const rateStars = document.querySelectorAll('.rate-product__star-icon');
-    let selectedStar;
-    let i;
-
-    for (i = 0; i < rateStars.length; i += 1) {
-        rateStars[i].addEventListener('click', selectRate);
+export default class RateProduct {
+    constructor(container) {
+        this.container = container;
+        this.APIURl = 'http://localhost:5003/rating';
+        this.rateBtn = this.container.querySelector('.rate-product__btn');
+        this.reviewText = this.container.querySelector('.rate-product__input');
+        this.rateStars = Array.from(this.container.querySelectorAll('.rate-product__star-icon'));
+        this.selectedStar;
+        this.ratingStars();
+        this.rateBtn.addEventListener('click', this.sendRreview);
     }
 
-    function selectRate() {
-        selectedStar = this.value;
+    ratingStars = () => {
```

Comment:
this could be written the same way as `sendReview`. Put the event listener into `constructor` and create a function that saves the value.

But overall, this rating doesn't work very well. If I click on a star and then click to write the review, I lose the selected stars visually. It should be seen what I selected even if you lose focus of that star.

Check our solution from Walmark Line, we used radio buttons. And then you could for example get the biggest value and send via API.